### PR TITLE
fix: changed height calc for embeded app page

### LINKED ISSUE #8573: [Bug]: Whitespace in iFrame viewport while embedding the app
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior

Using `embed=true` causes iframes to have a large gap at the bottom of the viewport

Screenshot 2021-10-17 at 11 19 17 AM

### Steps To Reproduce

```markdown
1. Open an online JS environment, repl.it / jsfiddle.net
2. Create an iFrame
3. Add Appsmith embed URL

Its observed that there is a white space b/w the page bottom and iframe bottom.
```

### Environment

Production

### Version

Cloud

### PR DESCRIPTION
**Pull request by techbhavin**
## Description

> changed height calc for embedded app page

Fixes #8573 , #8393 

## Type of change

- Bugfix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: bug/whitespace-embeded-page-bottom 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.01 **(0.01)** | 36.66 **(0.01)** | 34.83 **(0)** | 55.49 **(0)**
 :green_circle: | app/client/src/selectors/commentsSelectors.ts | 85.25 **(1.64)** | 64.71 **(2.95)** | 73.33 **(0)** | 90.59 **(2.35)**</details>

### CODE REVIEW

File: app/client/src/pages/AppViewer/index.tsx [hunk 1]
```diff
@@ -43,16 +43,19 @@ import AppViewerCommentsSidebar from "./AppViewerComemntsSidebar";
...
   justify-content: flex-start;
-  height: calc(
-    100vh -
-      ${(props) =>
-        !props.hasPages ? `${props.theme.smallHeaderHeight} - 1px` : "72px"}
-  );
+  ${(props) =>
+    props.isEmbeded
+      ? // embeded page will not have top header
+        "height: 100vh"
```
Comment (jsartisan) [inline app/client/src/pages/AppViewer/index.tsx:54]: This approach is Ok for now. But I went with a different one. I was calculating the header height and was storing it in redux. Then I over here, I was subtracting that header's height from the body's height.

**Review by jsartisan — APPROVED** [review]

**Comment by YogeshJayaseelan** [comment]
@techbhavin Verified this and it is working as expected

Screenshot 2022-01-18 at 3 15 46 PM

With respect to online JS environment we cannot test deploy previews due to authorization will not work in DP's is that okay? @techbhavin @dilippitchika

**Comment by somangshu** [comment]
> With respect to online JS environment we cannot test deploy previews due to authorization will not work in DP's is that okay? @techbhavin @dilippitchika

@YogeshJayaseelan having a hard time understanding this, can you please elaborate?

**Pull request merged**